for_012_fire_risk/contents/src/__init__.py

```diff
@@ -11,267 +11,466 @@
+    '''
+    Convert netcdf files to tifs
+    INPUT   files: list of file names for netcdfs that have already been downloaded (list of strings)
+    RETURN  tifs: list of file names for tifs that have been generated (list of strings)
+    '''
+    # create an empty list to store the names of the tifs we generate from all input netcdf files
     tifs = []
+    # go through each netcdf file and translate
     for f in files:
+        # create an empty list to store the names of the tifs we generate from this netcdf file
         band_tifs = []
+        # go through each variables to process in this netcdf file
         for sds_name in SDS_NAMES:
             # extract subdataset by name
             sds_path = sds_name.format(fname=f)
-            band_tif = '{}_{}.tif'.format(os.path.splitext(f)[0], sds_name.split('_')[-1]) #naming tiffs
-            cmd = ['gdal_translate','-q', '-a_nodata', str(NODATA_VALUE), '-a_srs', 'EPSG:4326', sds_path, band_tif] #'-q' means quiet so you don't see it
+            # generate a name to save the tif file we will translate the netcdf file's subdataset into
+            band_tif = '{}_{}.tif'.format(os.path.splitext(f)[0], sds_name.split('_')[-1]) 
+            # translate the netcdf file's subdataset into a tif
+            cmd = ['gdal_translate','-q', '-a_nodata', str(NODATA_VALUE), '-a_srs', 'EPSG:4326', sds_path, band_tif] 
             logging.debug('Converting {} to {}'.format(f, band_tif))
             subprocess.call(cmd)
+            # add the new subdataset tif files to the list of tifs generated from this netcdf file
             band_tifs.append(band_tif)
-        merged_tif = '{}.tif'.format(os.path.splitext(f)[0])  # naming tiffs
+        # generate a name to save the tif file that will be produced by merging all the sub tifs from this netcdf   
+        merged_tif = '{}.tif'.format(os.path.splitext(f)[0])  
+        # merge all the sub tifs from this netcdf to create an overall tif representing all variables
         merge_cmd = ['gdal_merge.py', '-seperate'] + band_tifs + ['-o', merged_tif]
         subprocess.call(merge_cmd)
+        # add the new tif files to the list of tifs
         tifs.append(merged_tif)
     return tifs
 
 def list_available_files(url, ext=''):
+    '''
+    Fetch a list of filenames from source url by year
+    INPUT   url: url for fire weather data in the format "base url/{year}" (string)
+            ext: specify we want netcdf files by providing an extension for filename (string)
+    RETURN  list of available filenames from source website for the input date's year (list of strings)
+    '''    
+    # open and read the url
     page = requests.get(url).text
+    # use BeautifulSoup to read the content as a nested data structure
     soup = BeautifulSoup(page, 'html.parser')
+    # Extract all the <a> tags within the html content to find the files available for download marked with these tags.
+    # Get only the files that ends with a netcdf extension ('.nc')
```

Comment:
This function is written to be general so it doesn't always check for netcdfs..it can check for any extension you give it
```suggestion
    # Get only the files that ends with input extension (if specified)
```